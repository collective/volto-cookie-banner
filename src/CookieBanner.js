import React from 'react'
import PropTypes from 'prop-types'
import CookieConsent from 'react-cookie-consent'

const CookieBanner = ({ text, buttonText }) => <CookieConsent buttonText={buttonText ?? 'OK'}>{text}</CookieConsent>

CookieBanner.propTypes = {
  text: PropTypes.string.isRequired,
  buttonText: PropTypes.string,
}

export default CookieBanner
