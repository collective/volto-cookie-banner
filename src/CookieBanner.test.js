import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render } from '@testing-library/react'
import { axe, toHaveNoViolations } from 'jest-axe'

import CookieBanner from './CookieBanner'

expect.extend(toHaveNoViolations)
afterEach(cleanup)

test('CookieBanner is being rendered', () => {
  const component = renderer.create(<CookieBanner text="cacca" />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('CookieBanner is accessible', async () => {
  const { container } = render(<CookieBanner text="cacca" />)
  const results = await axe(container)

  expect(results).toHaveNoViolations()
})
