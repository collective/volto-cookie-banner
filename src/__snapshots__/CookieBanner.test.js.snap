// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieBanner is being rendered 1`] = `
<div
  className="cookieConsent "
  style={
    Object {
      "alignItems": "baseline",
      "background": "#353535",
      "bottom": "0",
      "color": "white",
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "left": "0",
      "position": "fixed",
      "width": "100%",
      "zIndex": "999",
    }
  }
>
  <div
    className=""
    style={
      Object {
        "flex": "1 0 300px",
        "margin": "15px",
      }
    }
  >
    cacca
  </div>
  <button
    className=""
    id=""
    onClick={[Function]}
    style={
      Object {
        "background": "#ffd42d",
        "border": "0",
        "borderRadius": "0px",
        "boxShadow": "none",
        "color": "black",
        "cursor": "pointer",
        "flex": "0 0 auto",
        "margin": "15px",
        "padding": "5px 10px",
      }
    }
  >
    OK
  </button>
</div>
`;
